src/vs/workbench/services/preferences/common/preferences.ts

```diff
@@ -50,7 +63,7 @@ export interface ISetting {
 	deprecationMessage?: string;
 
 	scope?: ConfigurationScope;
-	type?: string | string[];
+	type?: SettingValueType | SettingValueType[];
```

Comment:
Possibly other fields should be changed as well?

Reply:
See comment below, is it possible to keep this as 'string' and only use SettingValueType inside the tree code?